src/server_manager/web_app/app.ts

```diff
@@ -373,18 +372,17 @@ export class App {
   }
 
   private connectToDigitalOcean() {
+    const oauthUi = this.appRoot.getAndShowDigitalOceanOauthFlow();
     const session = runDigitalOceanOauth();
-    this.appRoot
-        .showModalDialog(
-            'Awaiting authorization',  // Don't display any title.
-            'On the DigitalOcean window that was opened on your browser, sign in or create a new account, then authorize the Outline Manager application',
-            ['Cancel'])
-        .then(() => {
-          session.cancel();
-        });
+    const handleOauthFlowCanceled = () => {
+      session.cancel();
+      this.clearCredentialsAndShowIntro();
+      this.appRoot.removeEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
+    };
+    this.appRoot.addEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
```

Comment:
Same here. The listener may never be removed.

Reply:
Done.